azurelinuxagent/common/osutil/default.py

```diff
@@ -516,8 +516,15 @@ def set_dhcp_hostname(self, hostname):
                                       'send host-name',
                                       'send host-name "{0}";'.format(hostname))
 
-    def restart_if(self, ifname):
-        shellutil.run("ifdown {0} && ifup {1}".format(ifname, ifname))
+    def restart_if(self, ifname, retries=3, wait=5):
+        for attempt in range(1, retries+1, 1):
+            return_code=shellutil.run("ifdown {0} && ifup {1}".format(ifname, ifname))
+            if return_code == 0:
+                return
+            logger.warn("failed to restart {0}: return code {1}".format(ifname, return_code))
+            if attempt < retries:
+                logger.info("retrying in {0} seconds".format(wait))
+                time.sleep(wait)
```

Comment:
I wonder why they didn't save and return the result of `restart_if` in the first place. Does `shellutil.run` throw and exception (I guess not)? 

In any case it might be helpful to see that this method returns an exception or some error indicator when retries are exceeded and the agent fails accordingly. I'm not sure if it's OK to continue in this particular failure mode.